src/tests/suite/daos_container.c

```diff
@@ -771,6 +771,226 @@ co_set_prop(void **state)
 	test_teardown((void **)&arg);
 }
 
+static void
+co_create_access_denied(void **state)
+{
+	test_arg_t	*arg0 = *state;
+	test_arg_t	*arg = NULL;
+	daos_prop_t	*prop;
+	int		 rc;
+
+	rc = test_setup((void **)&arg, SETUP_EQ, arg0->multi_rank,
+			DEFAULT_POOL_SIZE, NULL);
+	assert_int_equal(rc, 0);
+
+	print_message("Try to create container on pool with no create perms\n");
+
+	/* on the pool, write is an alias for create+del cont */
+	prop = get_daos_prop_with_owner_acl_perms(DAOS_ACL_PERM_POOL_ALL &
+						  ~DAOS_ACL_PERM_CREATE_CONT &
+						  ~DAOS_ACL_PERM_WRITE,
+						  DAOS_PROP_PO_ACL);
+
+	while (!rc && arg->setup_state != SETUP_POOL_CONNECT)
+		rc = test_setup_next_step((void **)&arg, NULL, prop, NULL);
+	assert_int_equal(rc, 0);
+
+	MPI_Barrier(MPI_COMM_WORLD);
+
+	uuid_generate(arg->co_uuid);
+	rc = daos_cont_create(arg->pool.poh, arg->co_uuid, NULL, NULL);
```

Comment:
I wound up going back to rank 0 only for this one - this actually looked like it hung on CI and timed out waiting. And the logic shouldn't care about ranks anyway, so it is better to keep it simple I think.

Reply:
In the latest version of the patch the skip() call above will cause non rank 0 processes to return (and not call test_setup and test_setup_next_step particularly). However,those functions (when arg->multi_rank is true) have rank 0 perform an action, and all ranks participate in a collective MPI_Bcast() - for example to exchange the rank 0 rc value. So would these MPI_Bcast() calls then hang since the non rank 0 processes do not call MPI_Bcast()?

Reply:
Ahh, that may be what caused the hang I was seeing in CI. Good catch. I'll rework this and push up another revision. I only have a single node to test (where I can't see any of the multi-node issues) so I've been stuck guessing and pushing up to CI.